File: tests/dt_args_basic.rs
mod dt_test_utils;
use dt_test_utils::dt;

#[test]
fn status() {
    assert_eq!("[ ]\n", dt(&["status"]).stdout);
}

#[test]
fn one_plus_one_is_two() {
    // dt 1 1 + print
    let res = dt(&["1", "1", "+", "println"]);
    assert_eq!("2\n", res.stdout);
}

#[test]
fn quoted_one_plus_one_is_two() {
    // dt "1 1 + print"
    let res = dt(&["1 1 + println"]);
    assert_eq!("2\n", res.stdout);
}

File: src/tests/dt_args_basic.zig
const std = @import("std");
const expectEqualStrings = std.testing.expectEqualStrings;

const dt = @import("dt_test_utils.zig").dt;

test "status" {
    try expectEqualStrings("[ ]\n", (try dt(&.{ "quote-all", "drop", "status" })).stderr);
}

test "one_plus_one_is_two" {
    try expectEqualStrings("2\n", (try dt(&.{ "1", "1", "+", "println" })).stdout);
}

test "quoted_one_plus_one_is_two" {
    try expectEqualStrings("2\n", (try dt(&.{"1 1 + println"})).stdout);
}
